core/src/test/java/io/questdb/test/griffin/ParquetTest.java

```diff
@@ -275,4 +270,94 @@ public void testSinglePartition() throws Exception {
             );
         });
     }
+
+    @Test
+    public void testSymbols() throws Exception {
+        assertMemoryLeak(() -> {
+            ddl(
+                    "create table x (\n" +
+                            "  id int,\n" +
+                            "  ts timestamp,\n" +
+                            "  name symbol\n" +
+                            ") timestamp(ts) partition by day;"
+            );
+
+            // Day 0 -- using every symbol, two nulls
+            insert("insert into x values (0, 0, 'SYM_A')");
+            insert("insert into x values (1, 10000000, 'SYM_A')");
+            insert("insert into x values (2, 20000000, 'SYM_B_junk123')");
+            insert("insert into x values (3, 30000000, 'SYM_C_junk123123123123')");
+            insert("insert into x values (4, 40000000, 'SYM_D_junk12319993')");
+            insert("insert into x values (5, 50000000, 'SYM_E_junk9923')");
+            insert("insert into x values (6, 60000000, 'SYM_A')");
+            insert("insert into x values (7, 70000000, NULL)");
+            insert("insert into x values (8, 80000000, 'SYM_C_junk123123123123')");
+            insert("insert into x values (9, 90000000, NULL)");
+
+            // Day 1, NULLS, using two symbols
+            insert("insert into x values (10, 86400000000, 'SYM_B_junk123')");
+            insert("insert into x values (11, 86410000000, NULL)");
+            insert("insert into x values (12, 86420000000, 'SYM_B_junk123')");
+            insert("insert into x values (13, 86430000000, 'SYM_B_junk123')");
+            insert("insert into x values (14, 86440000000, 'SYM_B_junk123')");
+            insert("insert into x values (15, 86450000000, NULL)");
+            insert("insert into x values (16, 86460000000, NULL)");
+            insert("insert into x values (17, 86470000000, 'SYM_D_junk12319993')");
+            insert("insert into x values (18, 86480000000, NULL)");
+            insert("insert into x values (19, 86490000000, NULL)");
+
+            // Day 2, no nulls, using first and last symbols only
+            insert("insert into x values (20, 172800000000, 'SYM_A')");
+            insert("insert into x values (21, 172810000000, 'SYM_A')");
+            insert("insert into x values (22, 172820000000, 'SYM_A')");
+            insert("insert into x values (23, 172830000000, 'SYM_A')");
+            insert("insert into x values (24, 172840000000, 'SYM_E_junk9923')");
+            insert("insert into x values (25, 172850000000, 'SYM_E_junk9923')");
+            insert("insert into x values (26, 172860000000, 'SYM_E_junk9923')");
+
+            TestUtils.LeakProneCode checkData = () -> {
```

Comment:
nit: it's enough to use `Runnable` instead of custom `TestUtils.LeakProneCode` here

Reply:
Nope.
Runnable doesn't allow throwing an exception.

Reply:
I forgot about the exception in the signature. In that case, `CheckedRunnable` might do fine. I don't see why we have `LeakProneCode` while it's exactly the same thing as `CheckedRunnable`.